rfd/0178-app-custom-icons.md

```diff
@@ -0,0 +1,97 @@
+---
+author: Lisa Kim (lisa@goteleport.com)
+state: draft
+---
+
+# RFD 178 - Application Resource Custom Icon
+
+## Required Approvers
+
+- Engineering: @romant && @avatus && (someone from the scale team))
+
+## What
+
+Support custom icon uploads for resources through yaml. We can start with feature request [#4712](https://github.com/gravitational/teleport/issues/4712) where
+users provide a URL and Teleport's auth server will download the image and store it in the backend.
+
+## Why
+
+Users want to be able to recognize their applications by famaliar icons/logos rather than reading the application name.
+
+Today, the web UI has some hard coded methods to [guess](https://github.com/gravitational/teleport/blob/dc1199636558f85ed275512175a12207a8034d44/web/packages/shared/components/UnifiedResources/shared/viewItemsFactory.ts#L184) which icon to show for an application by looking at the application name (which may or may not be an accurate depiction of the application) and if nothing matches, we use a generic icon to mean `application`. The guess type is limited to a few handpicked applications.
+
+## Details
+
+### User Story
```

Comment:
Can we go into the user details a bit more?

- what happens if the user enters an invalid URL?
- what happens if the user enters a valid URL but we fail to connect to it?
- what happens if the user enters a valid URL but it serves up something other than an icon resource?
- is there any way for the user to preview what the icon looks like before it gets stored in the backend?